src/cpu/aarch64/jit_aarch64_sve_512_conv_kernel.hpp

```diff
@@ -0,0 +1,685 @@
+                        ptr(in, static_cast<int32_t>(VL_OFS(ofs))));
+            } else {
+                add_imm(reg_tmp_addr, in, ofs, reg_tmp_imm);
+                prfw(op_sve, reg_p_all_ones, ptr(reg_tmp_addr));
+            }
+        }
+    }
+
+#ifndef DISABLE_ELTWISE
+    jit_uni_eltwise_injector_f32<avx512_common> *eltwise_injector_;
+#endif // #ifndef DISABLE_ELTWISE
+
+    inline void prepare_output(int ur_w);
+    inline void store_output(int ur_w);
+    inline void compute_loop_fma_core(int ur_w, int pad_l, int pad_r);
+    inline void compute_loop(int ur_w, int pad_l, int pad_r);
+
+    void generate() override;
+
+    inline size_t get_output_offset(int oi, int n_oc_block) {
+        const bool is_nxc_layout = is_dst_layout_nxc();
+        size_t ow_str = is_nxc_layout ? jcp.ngroups * jcp.oc : jcp.oc_block;
+        size_t ocb_str = is_nxc_layout
+                ? jcp.oc_block
+                : (size_t)jcp.od * jcp.oh * jcp.ow * jcp.oc_block;
+
+        return jcp.typesize_out * (n_oc_block * ocb_str + oi * ow_str);
+    }
+
+    inline size_t get_input_offset(int ki, int ic, int oi, int pad_l) {
+        const bool is_nxc_layout = is_src_layout_nxc();
+        size_t iw_str = is_nxc_layout ? jcp.ngroups * jcp.ic
+                                      : (!jcp.is_1stconv ? jcp.ic_block : 1);
+        size_t ic_str = !jcp.is_1stconv || is_nxc_layout
+                ? 1
+                : (size_t)jcp.iw * jcp.ih * jcp.id;
+        size_t iw_idx = ki * (jcp.dilate_w + 1) + oi * jcp.stride_w - pad_l;
+
+        return jcp.typesize_in * (iw_idx * iw_str + ic * ic_str);
+    }
+
+    inline int get_kernel_offset(
+            int ki, int ic, int n_oc_block, int ker_number) {
+        return jcp.typesize_in * jcp.oc_block
+                * (n_oc_block * jcp.nb_ic * jcp.ic_block * jcp.kh * jcp.kw
+                                * jcp.kd
+                        + (ic + ker_number) + ki * jcp.ic_block);
+    }
+
+    inline int get_ow_start(int ki, int pad_l) {
+        return nstl::max(0,
+                utils::div_up(pad_l - ki * (jcp.dilate_w + 1), jcp.stride_w));
+    }
+
+    inline int get_ow_end(int ur_w, int ki, int pad_r) {
+        return ur_w
+                - nstl::max(0,
+                        utils::div_up(
+                                pad_r - (jcp.kw - 1 - ki) * (jcp.dilate_w + 1),
+                                jcp.stride_w));
+    }
+    inline bool is_src_layout_nxc() {
+        return utils::one_of(jcp.src_tag, format_tag::ndhwc, format_tag::nhwc,
+                format_tag::nwc);
+    }
+    inline bool is_dst_layout_nxc() {
+        return utils::one_of(jcp.dst_tag, format_tag::ndhwc, format_tag::nhwc,
+                format_tag::nwc);
+    }
+};
+
+struct jit_aarch64_sve_512_conv_fwd_kernel {
+
+    jit_aarch64_sve_512_conv_fwd_kernel(
+            const jit_conv_conf_t ajcp, const primitive_attr_t &attr)
+        : kernel_(nullptr) {
+        switch (ajcp.oc_block) {
+            case 16:
```

Comment:
nit: If there are no intentions to support other vector register size variants (!= 512bit), I would suggest removing the templated struct that encapsulates `_jit_aarch64_sve_512_conv_fwd_kernel<16> `.

Reply:
Thank you for your suggestion. I will remove these.